src/machines/__tests__/auth-token-polling-machine.test.ts

```diff
@@ -42,25 +36,32 @@ test('it invokes the requestAuthToken service', () => {
   expect(mockedFunc).toHaveBeenCalled()
 })
 
-test('it assigns the authToken to context when received', async () => {
+test('it assigns the authToken to context when received', (done) => {
+  const clock = new SimulatedClock()
+
   const pollingService = interpret(
     authTokenPollingMachine.withConfig({
       services: {
         requestAuthToken: () => Promise.resolve({authToken: 'auth123'}),
       },
     }),
-  )
+    {clock},
+  ).onTransition((state) => {
+    if (state.matches({pending: 'verifyRetrieval'})) {
+      clock.increment(4000)
+    }
+
+    if (state.matches('authTokenRetrieved')) {
```

Comment:
One of my hardline rules is "no logic in tests"

https://www.slideshare.net/joelhooks/unit-testing-guidelines

<img width="570" alt="Screen Shot 2021-11-09 at 3 28 55 PM" src="https://user-images.githubusercontent.com/86834/141022090-7f1e54e2-565d-43e7-b550-e0e5956b56c1.png">

If this condition isn't true the test passes, which doesn't seem like a situation we'd want.

I don't fully understand the clock overhead yet